src/Nethermind/Nethermind.Logging.NLog/NLogManager.cs

```diff
@@ -12,140 +12,147 @@
 using NLog.Targets;
 using Level = NLog.LogLevel;
 
-namespace Nethermind.Logging.NLog
+namespace Nethermind.Logging.NLog;
+
+public class NLogManager : ILogManager
 {
-    public class NLogManager : ILogManager
+    private const string DefaultFileTargetName = "file-async_wrapped";
+    private const string DefaultFolder = "logs";
+
+    public NLogManager(string logFileName, string logDirectory = null, string logRules = null)
     {
-        private const string DefaultFileTargetName = "file-async_wrapped";
+        Setup(logFileName, logDirectory, logRules);
+        // Required since 'NLog.config' could change during runtime, we need to re-apply the configuration
+        LogManager.ConfigurationChanged += (sender, args) => Setup(logFileName, logDirectory, logRules);
+    }
 
-        public NLogManager(string logFileName, string logDirectory = null, string logRules = null)
-        {
-            logDirectory = SetupLogDirectory(logDirectory);
-            SetupLogFile(logFileName, logDirectory);
-            SetupLogRules(logRules);
-        }
+    private void Setup(string logFileName, string logDirectory = null, string logRules = null)
+    {
+        logDirectory = SetupLogDirectory(logDirectory);
+        SetupLogFile(logFileName, logDirectory);
+        SetupLogRules(logRules);
+    }
```

Comment:
Only actual changes: We run the `Setup` on the constructor, and whenever NLog detects changes to the configuration.